exporter/otlphttpexporter/otlp_test.go

```diff
@@ -600,6 +597,128 @@ func TestPartialSuccess_metrics(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestEncoding(t *testing.T) {
+	set := exportertest.NewNopCreateSettings()
+	set.BuildInfo.Description = "Collector"
+	set.BuildInfo.Version = "1.2.3test"
+
+	tests := []struct {
+		name             string
+		encoding         EncodingType
+		expectedEncoding EncodingType
+	}{
+		{
+			name:             "default_encoding",
+			expectedEncoding: "application/x-protobuf",
+		},
+		{
+			name:             "explicit_proto_encoding",
+			encoding:         EncodingProto,
+			expectedEncoding: "application/x-protobuf",
+		},
+		{
+			name:             "json_encoding",
+			encoding:         EncodingJSON,
+			expectedEncoding: "application/json",
+		},
+	}
+
+	t.Run("traces", func(t *testing.T) {
+		for _, test := range tests {
+			t.Run(test.name, func(t *testing.T) {
+				srv := createBackend("/v1/traces", func(writer http.ResponseWriter, request *http.Request) {
+					assert.Contains(t, request.Header.Get("content-type"), test.expectedEncoding)
+					writer.WriteHeader(200)
+				})
+				defer srv.Close()
+
+				cfg := &Config{
+					TracesEndpoint: fmt.Sprintf("%s/v1/traces", srv.URL),
+					Encoding:       test.encoding,
+				}
+				exp, err := createTracesExporter(context.Background(), set, cfg)
+				require.NoError(t, err)
+
+				// start the exporter
+				err = exp.Start(context.Background(), componenttest.NewNopHost())
+				require.NoError(t, err)
+				t.Cleanup(func() {
+					require.NoError(t, exp.Shutdown(context.Background()))
+				})
+
+				// generate data
+				traces := ptrace.NewTraces()
+				err = exp.ConsumeTraces(context.Background(), traces)
+				require.NoError(t, err)
```

Comment:
Can you add tests on malformed data? Same for metrics and logs

Reply:
what do you mean by malformed data? invalid `encoding` field?

Reply:
I mean invalid proto or json text data, i.e. when err is not nil.

Reply:
Chatted offline, with the current setup it's difficult to generate data that cannot be parsed, so mark this as resolved.